### PR TITLE
Added generic shield for ST7789V display controller

### LINKED ISSUE #20278: Something is wrong when trying ST7789V sample
**Describe the bug**
Hello,
I have a ST7789V based LCD module ( a 1.54" one, not TL019FQV01), and nrf52_pca10040 board,  so I did a try on ST7789V sample.
Unfortunately,  I got "display_st7789v: Could not get SPI device for LCD", I checked the code and realize that it can't find it from the inited driver.
I don't understand why, any ideas?

**To Reproduce**
Just follow the readme in ST7789V sample

**Expected behavior**
see something being showed on the LCD

**Impact**
annoyance

**Screenshots or console output**
***** Booting Zephyr OS build zephyr-v2.0.0-1112-gb71eb71de72f *****
[00:00:00.005,462] <err> display_st7789v: Could not get SPI device for LCD
[00:00:00.005,462] <inf> app: ST7789V display sample
[00:00:00.005,493] <err> app: Device not found. Aborting test.

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain: Zephyr SDK
 - Commit SHA or Version used: 2.099

**Additional context**
nothing more by now

### PR DESCRIPTION
**Pull request by vanwinkeljan**
Added generic shield for ST7789V display controller

Fixes: #20278

### CODE REVIEW

File: boards/shields/st7789v_generic/doc/index.rst [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+Generic ST7789V Display Shield
+##############################
+
+Overview
+********
+
+This is a generic shield for display shields based on ST7789V display
+controller. More information about the controller can be found in
+`ST7789V Datasheet_`.
```
Comment (dbkinder) [inline boards/shields/st7789v_generic/doc/index.rst]: Underscore (indicating an outgoing link) is after the back-quoted text:
```
`ST7789V Datasheet`_.
```

File: boards/shields/st7789v_generic/doc/index.rst [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+Pins Assignment of the Generic ST7789V Display Shield
+=====================================================
+
++-----------------------+---------------------------------------------+
+| Arduino Connector Pin | Function                                    |
++=======================+=============================================+
++-----------------------+---------------------------------------------+
```
Comment (dbkinder) [inline boards/shields/st7789v_generic/doc/index.rst]: delete this line (it creates an unnecessary blank row in the generated HTML)

File: boards/shields/st7789v_generic/doc/index.rst [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+| D12                   | SPI MISO         (Serial Data Out)          |
++-----------------------+---------------------------------------------+
+| D13                   | SPI SCK          (Serial Clock Input)       |
++-----------------------+---------------------------------------------+
+
+Requirements
+************
+
+This shield can only be used with a board which provides a configuration
```
Comment (dbkinder) [inline boards/shields/st7789v_generic/doc/index.rst]: change ``which`` to ``that``

**Review by dbkinder — CHANGES_REQUESTED** [review]

File: boards/shields/st7789v_generic/doc/index.rst (hunk 1 above)
Comment (vanwinkeljan) [reply boards/shields/st7789v_generic/doc/index.rst]: thx for the input!

**Comment by jfischer-phytec-iot** [comment]
@vanwinkeljan I do not have TL019FQV01 display but the [Waveshare 1.3inch 240x240](https://www.waveshare.com/1.3inch-LCD-Module.htm). TL019FQV01 does not seem to be so common and I think it should not be the default one. I try to find some time to add support for the Waveshare display. Also the [Adafruit 1.3" 240x240](https://www.adafruit.com/product/4313) seems to be identical to one from Waveshare.

File: boards/shields/st7789v_generic/doc/index.rst [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+Pins Assignment of the Generic ST7789V Display Shield
+=====================================================
+
++-----------------------+---------------------------------------------+
+| Arduino Connector Pin | Function                                    |
++=======================+=============================================+
+| D8                    | ST7789V Reset                               |
++-----------------------+---------------------------------------------+
+| D9                    | ST7789V DC       (Data/Command)             |
```
Comment (dbkinder) [inline boards/shields/st7789v_generic/doc/index.rst]: While this is OK, if you really want to align the description part (in parentheses), you should add another column under the heading bar:
```
+-----------------------+---------------------------------------------+
| Arduino Connector Pin | Function                                    |
+=======================+================+============================+
| D8                    | ST7789V Reset  |                            |
+-----------------------+----------------+----------------------------+
| D9                    | ST7789V DC     | (Data/Command)             |
```
etc.

**Review by dbkinder — APPROVED** [review]
LGTM, with a potential table format edit.

**Comment by jfischer-phytec-iot** [comment]
draft for the Waveshare 1.3inch 240x240 - https://github.com/jfischer-phytec-iot/zephyr/tree/st7789v_240x240

**Comment by vanwinkeljan** [comment]
@jfischer-phytec-iot do you want me to take along the waveshare support or will you push this through a separate PR?

**Comment by jfischer-phytec-iot** [comment]
@vanwinkeljan see #20778

**Review by jfischer-no — APPROVED** [review]

**Comment by vanwinkeljan** [comment]
@jfischer-phytec-iot I assume you tested this on Waveshare as such I removed the label `blocked: HW test`?

**Comment by jfischer-phytec-iot** [comment]
> @jfischer-phytec-iot I assume you tested this on Waveshare as such I removed the label `blocked: HW test`?

Yes, I did

**Comment by jfischer-phytec-iot** [comment]
@erwango please review

**Review by erwango — CHANGES_REQUESTED** [review]
I'm fine with the change despite conflict with https://github.com/zephyrproject-rtos/zephyr/pull/20934.
Let's see which one goes in first and which one has to rebase.

**Comment by vanwinkeljan** [comment]
> Let's see which one goes in first and which one has to rebase.

Looks like #20934 as you request changes on this one ;)

**Comment by erwango** [comment]
> Looks like #20934 as you request changes on this one ;)

Oups! Seems I clicked the wrong button.. approved!

**Review by erwango — APPROVED** [review]

File: samples/display/st7789v/src/main.c [hunk 5]
```diff
@@ -121,6 +121,7 @@ void main(void)
 			*(buf + idx + 1) = (color_rgb >> 0) & 0xFFU;
 		}
 #else
+		int color = cnt % 4;
```
Comment (MaureenHelm) [inline samples/display/st7789v/src/main.c]: is this a stray change? it doesn't seem related to the commit message `samples: display: Use ST7789V generic shield`

Comment (vanwinkeljan) [reply samples/display/st7789v/src/main.c]: Looks like it, but it is a bug fix for the sample in case RGB565 is not defined

Comment (jfischer-phytec-iot) [reply samples/display/st7789v/src/main.c]: Not nice but this sample (and any other controller type specific) should be removed after we merged #20580

**Comment by jfischer-phytec-iot** [comment]
@MaureenHelm @nashif @carlescufi please revise/merge
